crates/red_knot_test/src/parser.rs

```diff
@@ -465,11 +484,23 @@ impl<'s> Parser<'s> {
             }
         };
 
+        let mut snapshot_diagnostic = false;
+        for &(key, _value) in keyvals {
+            match key {
+                "snapshot-diagnostic" => snapshot_diagnostic = true,
+                unk => bail!(
+                    "Unrecognized config in code-block metadata. Found \
+                     key={unk:?}. Supported keys are: `snapshot-diagnostic`."
```

Comment:
hrm, it feels like we'll probably forget to update this error message at some point if we do end up adding more supported keyvals. Not that I have a better suggestion :/

Reply:
Yup, definitely a downside. I ended up deleting this code anyway.